contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
 			}
 
 			pod := p.(*Pod)
-			if recoverAssignedSlave(pod.Pod) != "" {
+			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
```

Comment:
If it's deleted before it's scheduled can it possibly enter into graceful
termination?

On Mon, Sep 21, 2015 at 1:24 PM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/scheduler/plugin.go
> <https://github.com/kubernetes/kubernetes/pull/13966#discussion_r39998547>
> :
>
> > @@ -437,7 +437,7 @@ func (q *queuer) Run(done <-chan struct{}) {
> >  			}
> >
> >  			pod := p.(*Pod)
> > -			if recoverAssignedSlave(pod.Pod) != "" {
> > +			if recoverAssignedSlave(pod.Pod) != "" || pod.InGracefulTermination() {
>
> Maybe we can simplify this one. The idea is that a pod is deleted before
> being scheduled (e.g. because no offer matched). I wonder who will delete
> such a pod though. Normally the kubelet has the logic to actually delete a
> pod after termination.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13966/files#r39998547>.
>